src/crypto/mod.rs

```diff
@@ -45,6 +48,66 @@ pub struct SigData {
     pub q2: [u8; 384],
 }
 
+/// EC KT-I Public Key, the x-coordinate followed by
+/// the y-coordinate (on the RFC 6090P-256 curve),
+/// 2 x 32 bytes.
+/// A.4, Table 7
+#[derive(Clone, Debug)]
+#[repr(C)]
+pub struct EcdsaPubKey {
+    pub x: [u8; 32],
+    pub y: [u8; 32],
+}
+
+impl EcdsaPubKey {
+    /// Cast an instance from a byte slice.
+    pub fn cast(raw: &[u8; size_of::<Self>()]) -> &Self {
+        unsafe { transmute(raw) }
+    }
+}
+
+/// ECDSA signature, the r component followed by the
+/// s component, 2 x 32 bytes.
+/// A.4, Table 6
+#[derive(Clone, Debug)]
+#[repr(C)]
+pub struct EcdsaP256Sig {
+    pub r: [u8; 32],
+    pub s: [u8; 32],
+}
+
+impl EcdsaP256Sig {
+    // The DER encoded length determined via es.encoded_len().
+    pub const DER_ENCODED_LEN: usize = 72;
```

Comment:
Why a constant? If someone has the signature can they calculate the length themselves?

Reply:
Yeah I was going to just use lazy static but then I have to make a new vec every time I want to verify the signature. I can do that if you think it would be an improvement

Reply:
https://github.com/RustCrypto/formats/blob/master/der/src/length.rs

Reply:
This is a neat way to satisfy both std and no-std developers with a maximally performant API, but I agree with Nathaniel that's probably overkill here that this can probably just return the fixed-size array.